### ISSUE TITLE
Diagrammes

### ISSUE THREAD

**Issue by Lastshot97** [maoschanz/slightstone-server#1]
![serverclassdiagram](https://user-images.githubusercontent.com/25321997/49285956-41728d00-f499-11e8-8eb7-dcec843c3037.png)

```
{
    "joueurCourant": "Arthur",
    "joueur": {
        "pseudo": "Arthur",
        "heros": {
            "type": "Guerrier",
            "image": "http://www.slightstone.com/images/guerrier.png",
            "pointsVie": 30,
            "pointsMana": 4,
            "pointsArmure": 0
        },
        "etatMain": [
            {
                "idCarte": 2,
                "etat": "New - Used",
                "nom": "Sanglier brocheroc", 
                "description": "...", 
                "image": "http://www.slightstone.com/images/sanglierBrocheroc.png",
                "mana": 1, 
                "dégats": 1,
                "vie": 1
            },
            {
                "idCarte": 3, 
                "etat": "New - Used",
                "nom": "Yéti Noroit", 
                "description": "...", 
                "image": "http://www.slightstone.com/images/yetiNoroit.png",
                "mana": 4, 
                "dégats": 4,
                "vie": 5
            }
        ],
        "etatBoard": [
            {
                "idCarte": 1, 
                "etat": "New - Updated - Dead",
                "nom": "Soldat du comté-de-l'or", 
                "description": "...", 
                "image": "http://www.slightstone.com/images/yetiNoroit.png",
                "mana": 1, 
                "dégats": 1,
                "vie": 2
            }
        ]
    },
    "adversaire": {
        "pseudo": "Charlotte",
        "heros": {
            "type": "Paladin",
            "image": "http://www.slightstone.com/images/paladin.png",
            "pointsVie": 30,
            "pointsMana": 6,
            "pointsArmure": 1
        },
        "nbCartesMain": 4,
        "etatBoard": [
            {
                "idCarte": 56, 
                "etat": "New - Updated - Dead",
                "nom": "Chevaucheur de loup", 
                "description": "...", 
                "image": "http://www.slightstone.com/images/chevaucheurDeLoup.png",
                "mana": 3, 
                "dégats": 3,
                "vie": 1
            }
        ]
    }
}
```

**Comment by maoschanz** [comment]
![cartoonist-alex-norris-is-the-artist-to-watch-out-for-at-mcm-comic-con-_-the-wharf4](https://user-images.githubusercontent.com/19996113/49529478-61011f80-f8b6-11e8-8e83-dadb881415db.jpg)

**Comment by maoschanz** [comment]
Dans le JSON, les cartes de la main devraient avoir un attribut booléen "needCible". Le héros, dont l'action peut requérir une cible aussi, aura besoin d'un attribut du même genre.

Il faut aussi qu'on sache ce que le client retourne au serveur

**Comment by Lastshot97** [comment]
ouais, il faudra ajouter ces deux informations : le booléen et le type de
cible autorisé

Le ven. 7 déc. 2018 à 10:31, Romain F. T. <notifications@github.com> a
écrit :

**Comment by maoschanz** [comment]
Le modèle n'est pas clair sur la gestion de la mana, il y a le truc stockmana lié au joueur ET l'attribut pointMana du héros, lequel fait quoi ?

**Comment by Lastshot97** [comment]
L'attrubut du héros ça correspond au coût de son attaque spéciale. Il
faudrait peut-être le renommé !

Le ven. 7 déc. 2018 à 11:03, Romain F. T. <notifications@github.com> a
écrit :

**Comment by maoschanz** [comment]
coutMana du coup

**Comment by Lastshot97** [comment]
Tu aimes vraiment factoriser les choses le plus possible xD
Oui on pourrait voir ça comme ça. C'est pas très représentatif du monde
réel mais ça marcherait.

Le ven. 7 déc. 2018 à 11:27, Romain F. T. <notifications@github.com> a
écrit :

**Comment by maoschanz** [comment]
C'est surtout stockable avec JPA :)

**Comment by Lastshot97** [comment]
Je viens d'éditer le json mais je n'ai pas encore mis à jours le diagramme.
J'ai également commit ce que j'avais fais, j'ai implémenté les fonctions
principales.
Désolé d'avoir tardé à commit, j'avais fait ça avant les fêtes et j'ai
complètement oublié de
commit.
Si tu as des questions n'hésite pas.

**Comment by maoschanz** [comment]
Ah zut j'avais fait 2 ou 3 trucs aussi, j'espère que ça va pas me faire des conflits

**Comment by Lastshot97** [comment]
Il y a de forte chance, si tu rencontres un pb n'hésite pas à ouvrir une
issue ou quelque chose du genre si tu veux que je regarde !

Le mar. 1 janv. 2019 à 22:35, Romain F. T. <notifications@github.com> a
écrit :

**Comment by maoschanz** [comment]
(Et AppTest, je suppose)

**Comment by Lastshot97** [comment]
Salut, en effet j'ai dû un peu rush le code, Arthur et Charlotte qui
travail sur le côté client commençait à s'impatienter de ne pas avoir le
serveur pour finir leur partie.
Je vais m'occuper de mettre le diagramme à jour dans la soirée.
En ce qui concerne les tests, il y a les classes Jeu et Joueur, peut-être
la classe Contrôleur également (c'est elle qui s'assure que les règles du
jeu sont respectées donc ça peut être bien qu'elle soit testée).
Ensuite, est-ce qu'on test la façade, personnellement j'ai pas très envie,
à voir si c'est important ^^

**Comment by maoschanz** [comment]
Il y a une raison particulière pour laquelle il y a des tests commentés dans TestActionAttaque ?

**Comment by Lastshot97** [comment]
Oui, il faut les changer de classe. J'ai déplacé la méthode récupérerCibles
dans la classe Jeu dans tester cette méthode dans la classe ActionAttaque
n'a plus de sens.
